processing/src/main/java/io/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -729,6 +768,18 @@ public boolean matches()
         }
       };
     }
+
+    private ValueMatcher makeLongValueMatcher(String dimension, DruidLongPredicate predicate)
+    {
+      // __time is the only long column currently supported
```

Comment:
Or, keep the special handling for time, but actually do support getting non-time longs (e.g. from aggregators). That'd be reasonable if there were no rollup.